addons/payment_worldline/models/payment_transaction.py

```diff
@@ -268,7 +268,7 @@ def _process_notification_data(self, notification_data):
 
         # Update the provider reference.
         payment_data = notification_data['payment']
-        self.provider_reference = payment_data.get('id', '').removesuffix('_0')
+        self.provider_reference = payment_data.get('id', '').split('_', 1)[0]
```

Comment:
If you expect the `id` to contain more than one `_` (according to your `maxsplit=1` param), you should join all the strings resulting of a normal split except the last one. Here, you split on the first occurrence of `_`.

Reply:
I'll change it as it makes sense, but we don't expect more than one AFAIK